### PR TITLE
add depends_on :python for older OS

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

**Comment by xu-cheng** [comment]
@tdsmith Mind to take a look on the bot failure? Thanks.

File: matplotlib.rb [hunk 1]
```diff
@@ -46,6 +46,7 @@ class Matplotlib < Formula
   end
 
   option "without-python", "Build without python2 support"
+  depends_on :python if MacOS.version <= :snow_leopard && build.with?("python")
```
Comment (tdsmith) [inline matplotlib.rb]: let's make this `depends_on :python => :recommended if MacOS.version <= :snow_leopard`

**Comment by tdsmith** [comment]
Dunno why things are failing on Mountain Lion; not enough info in the bot output to diagnose quickly. Looks bogus though.

**Comment by xu-cheng** [comment]
Updated and re-run the test. Still failing on Mountain Lion.

**Comment by tdsmith** [comment]
:shipit:

**Pull request closed without merging**